src/mechanics/solver/Solve.cpp

```diff
@@ -0,0 +1,37 @@
+#include "Solve.h"
+#include "EigenSolver.h"
+
+bool isPrefix(std::string input, std::string prefix)
+{
+    return input.compare(0, prefix.length(), prefix) == 0;
+}
+
+
+namespace NuTo
+{
+
+Eigen::VectorXd Solve(Eigen::SparseMatrix<double> A, Eigen::VectorXd b, std::string solver)
+{
+    if (isPrefix(solver, "Eigen"))
+        return EigenSolver(A, b, solver);
+    throw Exception("Que?");
+}
+
+
+GlobalDofVector Solve(GlobalDofMatrixSparse K, GlobalDofVector f, Constraint::Constraints bcs, DofType dof,
+                      int numIndependentDofs, double time, std::string solver)
```

Comment:
1) `numIndependentDofs`. Well. It may sound misleading that the constraints do not know the number of dofs. The numbering is done by `NuTo::DofNumbering`. So this is seems to be a conceptual flaw. Maybe having `Constraints::BuildConstraintMatrix(...)` is in the wrong class. Or the dof numbering should belong directly to the `Constraints` class. But since they _do not have to_, I chose to keep both things separated. Maybe there should be another class that handles both. Especially, since building the constraint matrix without a numbering is pointless... I'll think about that.

2) `time` could also defaulted, to `0.`

Reply:
1. This wasn't a complaint, just an observation. Maybe I could get `numIndependentDofs` from the matrix?
2. That doesn't solve the problem imho.

Reply:
1.) Yeah right. `f.J[dof].rows()` should do.